src/app/Shared/Dialog.component.tsx

```diff
@@ -10,11 +12,19 @@ export function DialogComponent(props: {
     dialogTitle: string;
     dialogContentText: JSX.Element;
     dialogContent: JSX.Element;
-    dialogButtons: JSX.Element;
+    cancelButton: JSX.Element | string;
+    submitButton: JSX.Element | string;
+    disableSubmitButton: boolean;
     onClose: () => void;
+    onCancelClick: () => void;

```

Comment:
You can remove the onClose() event completely here and in the parent components, if you merge the cancel and close event (they have the same meaning). So here it would be:
```
onCancel: () => void;
onSubmit: () => void;
...
handleClose = () => props.onCancel();
handleClickCancel = () => props.onCancel();
handleClickSubmit = () => props.onSubmit();
```

Reply:
Done.